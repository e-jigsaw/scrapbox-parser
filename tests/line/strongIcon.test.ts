describe('strongIcon', () => {
  it('Simple root strong icon', () => {
    expect('[[/icons/+1.icon]]').toMatchSnapshotWhenParsing({ hasTitle: false })
  })

  it('Simple relative strong icon', () => {
    expect('[[me.icon]]').toMatchSnapshotWhenParsing({ hasTitle: false })
  })

  it('Multiple icons', () => {
    expect('[[me.icon*3]]').toMatchSnapshotWhenParsing({ hasTitle: false })
  })

  it('strong icon and internal link on same line', () => {
    expect('[Internal link][[me.icon]]').toMatchSnapshotWhenParsing({ hasTitle: false })
  })
})
