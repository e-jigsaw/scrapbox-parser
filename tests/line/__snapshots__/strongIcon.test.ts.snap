// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`strongIcon Multiple icons: toMatchSnapshotWhenParsing 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "path": "me",
        "pathType": "relative",
        "type": "strongIcon",
      },
      Object {
        "path": "me",
        "pathType": "relative",
        "type": "strongIcon",
      },
      Object {
        "path": "me",
        "pathType": "relative",
        "type": "strongIcon",
      },
    ],
    "type": "line",
  },
]
`;

exports[`strongIcon Simple relative strong icon: toMatchSnapshotWhenParsing 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "path": "me",
        "pathType": "relative",
        "type": "strongIcon",
      },
    ],
    "type": "line",
  },
]
`;

exports[`strongIcon Simple root strong icon: toMatchSnapshotWhenParsing 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "path": "/icons/+1",
        "pathType": "root",
        "type": "strongIcon",
      },
    ],
    "type": "line",
  },
]
`;

exports[`strongIcon strong icon and internal link on same line: toMatchSnapshotWhenParsing 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "content": "",
        "href": "Internal link",
        "pathType": "relative",
        "type": "link",
      },
      Object {
        "path": "me",
        "pathType": "relative",
        "type": "strongIcon",
      },
    ],
    "type": "line",
  },
]
`;
